packages/grafana-data/src/transformations/transformers/limit.ts

```diff
@@ -18,21 +19,32 @@ export const limitTransformer: DataTransformerInfo<LimitTransformerOptions> = {
     limitField: DEFAULT_LIMIT_FIELD,
   },
 
-  operator: (options) => (source) =>
+  operator: (options, ctx) => (source) =>
     source.pipe(
       map((data) => {
-        const limitFieldMatch = options.limitField || DEFAULT_LIMIT_FIELD;
+        let limit = DEFAULT_LIMIT_FIELD;
+        if (options.limitField !== undefined) {
```

Comment:
```suggestion
        if (options.limitField !== undefined && typeof options.limitField === 'string') {
```
to remove a level of nested ifs?

Reply:
That second if clause has an else clause that needs to happen if `options.limitField` is a number, but shouldn't happen if it's undefined.